src/App.js

```diff
@@ -1,18 +1,114 @@
 import React, { Component } from 'react';
-import logo from './logo.svg';
 import './App.css';
 
+import { RecipeList } from './components/RecipeList';
+import { SearchBar } from './components/SearchBar';
+
+import { APP_ID, APP_KEY } from './config/';
+
+const ErrorMessage = ({message, isLoading}) => {
+  return (
+    <div className="error-message">
+      {
+        isLoading ? "Loading recipes..." : message
+      }
+    </div>
+  );
+};
+
 class App extends Component {
+  constructor(){
+    super();
+    this.state = {
+      searchTerm: '',
+      recipes: {
+        fetchedList: [],
+        errorMessage: ''
+      },
+      isLoading: false
+    }
+  }
+
+  updateSearchTerm = (e) => {
+    this.setState({
+      searchTerm: e.target.value
+    });
+  }
+
+  onClickSearch = (e) => {
+    e.preventDefault();
+    this.fetchRecipes(this.state.searchTerm);
+  }
+
+  fetchRecipes = (searchTerm) => {
+    const BASE_ENDPOINT = `https://api.edamam.com/search?q=${searchTerm}&app_id=${APP_ID}&app_key=${APP_KEY}`;
+
+    this.setState({
+      isLoading: true
+    });
+
+    fetch(BASE_ENDPOINT)
+    .then(res => res.json())
+    .then(recipes => {
+      if(recipes.hits.length > 0) {
+        const data = recipes.hits.map(({recipe}) => {
```

Comment:
Nice use of map!

Remember that passing in your adaption function is good for code reuse. Also, you don't need Object.assign here, but I like that you're making sure your being immutible.

You could do:
```
return {
 name: recipe.label,
 calories: Math.floor(recipe.calories),
 image: recipe.image,
 url: recipe.url
}
```
which returns a new object (without mutations)